test/integration/driver.go

```diff
@@ -126,18 +167,127 @@ func main() {
 	check(w.Flush())
 	check(f.Close())
 
-	run("kubectl apply -f " + yaml + " -n " + namespace)
-	pods := getPods()
-	log.Println("pods:", pods)
-	if dump {
-		dumpProxyLogs(pods["a"])
-		dumpProxyLogs(pods["b"])
-	}
+	run("kubectl apply -f " + simpleAppYaml + " -n " + namespace)
+}
+
+func setupVersionedApp() {
+	// write template
+	f, err := os.Create(versionedAppYaml)
+	check(err)
+	w := bufio.NewWriter(f)
+
+	check(write("test/integration/http-service-versions.yaml.tmpl", map[string]string{
+		"hub":     hub,
+		"tag":     tag,
+		"name":    "hello",
+		"service": "hello",
+		"port1":   "8080",
+		"port2":   "80",
+		"version": "v1",
+	}, w))
+
+	check(write("test/integration/http-service-versions.yaml.tmpl", map[string]string{
+		"hub":     hub,
+		"tag":     tag,
+		"name":    "world-v1",
+		"service": "world",
+		"port1":   "80",
+		"port2":   "8000",
+		"version": "v1",
+	}, w))
+
+	check(write("test/integration/http-service-versions.yaml.tmpl", map[string]string{
+		"hub":     hub,
+		"tag":     tag,
+		"name":    "world-v2",
+		"service": "world",
+		"port1":   "80",
+		"port2":   "8000",
+		"version": "v2",
+	}, w))
+
+	check(w.Flush())
+	check(f.Close())
+
+	run("kubectl apply -f " + versionedAppYaml + " -n " + namespace)
+}
+
+func checkBasicReachability(pods map[string]string) {
+	log.Printf("Verifying basic reachability across pods/services (a, b, and t)..")
 	ids := makeRequests(pods)
-	log.Println("requests:", ids)
+	if verbose {
+		log.Println("requests:", ids)
+	}
 	checkAccessLogs(pods, ids)
 	log.Println("Success!")
-	cleanup()
+}
+
+func checkRouting(pods map[string]string) {
+	// First test default routing
+	// Create a bytes buffer to hold the YAML form of rules
+	log.Println("Routing all traffic to world-v1 and verifying..")
+	var defaultRoute bytes.Buffer
+	w := bufio.NewWriter(&defaultRoute)
+
+	check(write("test/integration/rule-default-route.yaml.tmpl", map[string]string{
+		"destination": "world",
+		"namespace":   namespace,
+	}, w))
+
+	check(w.Flush())
+	check(addRule(defaultRoute.Bytes(), "route-rule", "default-route", namespace))
+	verifyRouting(pods, "hello", "world", 100, map[string]int{
+		"v1": 100,
+		"v2": 0,
+	})
+	log.Println("Success!")
+
+	log.Println("Routing 75 percent to world-v1, 25 percent to world-v2 and verifying..")
+	// Create a bytes buffer to hold the YAML form of rules
+	var weightedRoute bytes.Buffer
+	w = bufio.NewWriter(&weightedRoute)
+
+	check(write("test/integration/rule-weighted-route.yaml.tmpl", map[string]string{
+		"destination": "world",
+		"namespace":   namespace,
+	}, w))
+
+	check(w.Flush())
+	check(addRule(weightedRoute.Bytes(), "route-rule", "weighted-route", namespace))
```

Comment:
can we use the constant RouteRule for "route-rule"? This would prevent accidental failure in the future. It's probably right not to use the CLI interface and use API directly.